packages/cloud_firestore/cloud_firestore/lib/src/field_value.dart

```diff
@@ -53,13 +51,16 @@ class FieldValue extends FieldValuePlatform {
   dynamic _delegate;
 
   @override
+  // ignore: no_runtimetype_tostring
   String toString() => '$runtimeType($_delegate)';
 
   @override
-  bool operator ==(Object o) {
-    return o is FieldValue && o._delegate == _delegate;
+  // ignore: avoid_equals_and_hash_code_on_mutable_classes
```

Comment:
can't we make this class immutable?

Reply:
I could but then I get another warning:`must_be_immutable`.`FieldValue._delegate` is not final. I didn't want to change underlying implementations in case of unforeseen side effects.

Reply:
Given that `CollectionReference` was also changed to be immutable I don't see any issue with this one being changed either, _delegate doesn't get re-assigned either so making it final should be an issue.

![image](https://user-images.githubusercontent.com/5347038/106458823-d530b100-6488-11eb-84b8-63913ff93152.png)